Sources/Apollo/JSONStandardTypeConversions.swift

```diff
@@ -2,10 +2,14 @@ import Foundation
 
 extension String: JSONDecodable, JSONEncodable {
   public init(jsonValue value: JSONValue) throws {
-    guard let string = value as? String else {
-      throw JSONDecodingError.couldNotConvert(value: value, to: String.self)
+    switch value {
+    case let string as String:
+        self = string
+    case let int as Int:
```

Comment:
Is it the `case let` that allows this to only need `as` instead of `as?`?

Reply:
Yes, instead of throw if the value isn't string I want to have better granularity when managing the type. If the value is a `String` keep it as it, if is an `Int` cast it to `String` because is a custom scalar and we want to parse it properly in our app (specific case of the Timestamp described in the PR description). This of course could be extended to other basic types, like Doubles, Bool, etc.

Reply:
Yeah i think for now let's keep it to `String` and `Int` - my recollection is that we recommend explicitly against doing that for Bool scalars. In practice `Int` is really the only one I've seen people requesting.